pkg/minibroker/minibroker.go

```diff
@@ -506,77 +513,170 @@ func (c *Client) connectTiller() (*tiller.Client, func(), error) {
+		if err != nil {
+			return err
+		}
+		if len(services.Items) == 0 {
+			return osb.HTTPStatusCodeError{StatusCode: http.StatusNotFound}
+		}
+
+		secrets, err := c.coreClient.CoreV1().Secrets(releaseNamespace).List(filterByInstance)
+		if err != nil {
+			return err
+		}
+		if len(secrets.Items) == 0 {
+			return osb.HTTPStatusCodeError{StatusCode: http.StatusNotFound}
+		}
+
+		data := make(map[string]interface{})
+		for _, secret := range secrets.Items {
+			for key, value := range secret.Data {
+				data[key] = string(value)
+			}
+		}
+
+		// Apply additional provisioning logic for Service Catalog Enabled services
+		provider, ok := c.providers[serviceID]
+		if ok {
+			creds, err := provider.Bind(services.Items, params, data)
+			if err != nil {
+				return errors.Wrapf(err, "unable to bind instance %s", instanceID)
+			}
+			for k, v := range creds.ToMap() {
+				data[k] = v
+			}
+		}
+
+		// Record the result for later fetching
+		bindingResponse := osb.GetBindingResponse{
+			Credentials: data,
+			Parameters:  bindParams,
+		}
+		bindingResponseJSON, err := json.Marshal(bindingResponse)
+		if err != nil {
+			return err
+		}
+
+		err = c.updateConfigMap(instanceID, map[string]interface{}{
+			(BindingKeyPrefix + bindingID): string(bindingResponseJSON),
+		})
+		if err != nil {
+			return err
+		}
+
+		return nil
+	}()
+
+	operationState := osb.LastOperationResponse{}
+	if err == nil {
+		operationState.State = osb.StateSucceeded
+	} else {
+		glog.Errorf("Error binding instance %s: %s", instanceID, err)
+		operationState.State = osb.StateFailed
+		operationState.Description = strPtr(fmt.Sprintf("Failed to bind instance %q", instanceID))
+	}
+	operationStateJSON, err := json.Marshal(operationState)
 	if err != nil {
-		return nil, err
+		glog.Errorf("Error serializing bind operation state: %s", err)
+		return
 	}
-	if len(services.Items) == 0 {
-		return nil, osb.HTTPStatusCodeError{StatusCode: http.StatusNotFound}
+	updates := map[string]interface{}{
+		(BindingStateKeyPrefix + bindingID): string(operationStateJSON),
 	}
-
-	secrets, err := c.coreClient.CoreV1().Secrets(releaseNamespace).List(filterByInstance)
+	err = c.updateConfigMap(instanceID, updates)
 	if err != nil {
-		return nil, err
-	}
-	if len(secrets.Items) == 0 {
-		return nil, osb.HTTPStatusCodeError{StatusCode: http.StatusNotFound}
+		glog.Errorf("Error updating bind status: %s", err)
 	}
+}
 
-	data := make(map[string]interface{})
-	for _, secret := range secrets.Items {
-		for key, value := range secret.Data {
-			data[key] = string(value)
-		}
+// Unbind a previously-bound instance binding.
+func (c *Client) Unbind(instanceID, bindingID string) error {
+	// The only clean up we need to do is to remove the binding information
+	err := c.updateConfigMap(instanceID, map[string]interface{}{
+		(BindingStateKeyPrefix + bindingID): nil,
+		(BindingKeyPrefix + bindingID):      nil,
+	})
+	if err != nil {
+		return err
```

Comment:
Unless you are wrapping this error with a relevant message, this function can be written:

```
func (c *Client) Unbind(instanceID, bindingID string) error {
	// The only clean up we need to do is to remove the binding information
	return c.updateConfigMap(instanceID, map[string]interface{}{
		(BindingStateKeyPrefix + bindingID): nil,
		(BindingKeyPrefix + bindingID):      nil,
	})
}
```

Reply:
You're right; because `updateConfigMap` returns an `error` (and not a pointer to a struct that implements `error`), this is safe.  Fixed in 05d340faf1eeff23091b8d53be40f7bb72104011